### PR TITLE
Edit the embedded CLI example to reflect the new approach to running CLI apps on vercel

### PR DESCRIPTION
**Pull request by airhorns**
This follows up to https://github.com/gadget-inc/gadget/pull/5273 to add back a lightweight backend for the Shopify CLI app. The backend only really has one job: serve static files built by vite, but, crucially, with Shopify's security headers. To get those headers on Vercel, we need to use a Vercel serverless function, which means we can't get away with just serving the static assets that vite builds.

Followed the guide that we put together for a customer here: https://gist.github.com/airhorns/bd16f8b24424937e211e483a1475dd95

### CODE REVIEW

File: packages/shopify-cli-embedded/web/frontend/components/ProductManager.jsx [hunk 1]
```diff
@@ -1,33 +1,29 @@
...
-  const [{ data, fetching, error }, refresh] = useFindMany(api.shopifyProduct);
+  const [{ data, fetching, error }, _refresh] = useFindMany(api.shopifyProduct);
 
   if (error) return <>Error: {error.toString()}</>;
 
   return (
     <Layout>
       <Layout.Section>
-        {fetching && <span>Loading...</span>}
+        {fetching && <Spinner />}
```
Comment (rdraward) [inline packages/shopify-cli-embedded/web/frontend/components/ProductManager.jsx:13]: 👍

**Review by rdraward — APPROVED** [review]
Groovy, thanks @airhorns!

File: packages/product-bundles/web/frontend/vite.config.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ if (process.env.npm_lifecycle_event === "build" && !process.env.CI && !process.e
 
 const FRONTEND_PORT = 3000;
```
Comment (DevAOC) [inline packages/product-bundles/web/frontend/vite.config.js:13]: Do we still want this hardcoded or do we want to follow the process.env.FRONTEND_PORT examples we have in the docs?

**Review by DevAOC — APPROVED** [review]
One nitpick, otherwise its good to me

**Pull request merged**